test/v4/controllers/createUkPropertyPeriodSummary/def2/Def2_CreateUkPropertyPeriodSummaryValidatorSpec.scala

```diff
@@ -0,0 +1,530 @@
+                                       |            }
+                                       |        }
+                                       |    },
+                                       |    "ukNonFhlProperty": {
+                                       |        "income": {
+                                       |            "premiumsOfLeaseGrant": 42.12,
+                                       |            "reversePremiums": 84.31,
+                                       |            "periodAmount": 9884.93,
+                                       |            "taxDeducted": 842.99,
+                                       |            "otherIncome": 31.44,
+                                       |            "rentARoom": {
+                                       |                "rentsReceived": 947.66
+                                       |            }
+                                       |        },
+                                       |        "expenses": {
+                                       |            "premisesRunningCosts": 3123.21,
+                                       |            "repairsAndMaintenance": 928.42,
+                                       |            "financialCosts": 842.99,
+                                       |            "professionalFees": 8831.12,
+                                       |            "costOfServices": 484.12,
+                                       |            "other": 99282,
+                                       |            "residentialFinancialCost": 12.34,
+                                       |            "travelCosts": 974.47,
+                                       |            "residentialFinancialCostsCarriedForward": 12.34,
+                                       |            "rentARoom": {
+                                       |                "amountClaimed": 8842.43
+                                       |            }
+                                       |        }
+                                       |    }
+                                       |}
+                                       |""".stripMargin)
+
+
+  private val validMinimalFhlBody = validBody
+    .removeProperty("/ukNonFhlProperty")
+    .replaceWithEmptyObject("/ukFhlProperty")
+    .update(
+  "/ukFhlProperty",
+  Json.parse("""
+               | {
+               |   "income": {
+               |     "periodAmount": 5000.99
+               |   }
+               | }
+               |""".stripMargin)
+  )
+
+  private val validMinimalNonFhlBody = validBody
+    .removeProperty("/ukFhlProperty")
+    .replaceWithEmptyObject("/ukNonFhlProperty")
+    .update(
+  "/ukNonFhlProperty",
+  Json.parse("""
+               | {
+               |   "income": {
+               |     "periodAmount": 9884.93
+               |   }
+               | }
+               |""".stripMargin)
+  )
+
+  private val validBodyConsolidatedWithExtraFields = validBody
```

Comment:
This validBodyConsolidatedWithExtraFields should also have values for residentialFinancialCost and residentialFinancialCostsCarriedForward for ukNonFhlProperty.